### ISSUE TITLE
Application of regression head

### ISSUE THREAD

**Issue by jeffei** [center-for-humans-and-machines/transformer-heads#27]
They are such useful tools! Could you please share some excellent cases where regression or classification heads are used and perform better than LLM-based heads? The reason behind this request is that there are so many fine-tuning tasks built on LLM-based heads, such as sentiment classification and bounding box (bbox) detection tasks.

In my opinion, I intuitively hold the view that we should use numeric heads to obtain numeric values.

**Comment by yannikkellerde** [comment]
Hey, I don't have great comparison papers ready, but I can share a little bit of my perspective.
First of all, language-modelling heads are classification heads. They solve the vocab-size (e.g. 128256) class classfication task of predicting what token is most likely to come next after a sequence of text.

Thus, for a classification task like sentiment classification, there is nothing fundamentally wrong with using a language modelling head. While maybe more elegant and memory efficient, it does not really matter too much, if you train a dedicated 2-class classification head for positive/negative or if you just use the 128256-class language modelling head and only ever have the "positive" or "negative" token as your ground truth class labels. The cross-entropy loss of the 128256-class language modelling head becomes equivalent to the binary cross-entropy loss of a 2-class classifcation head in this case.

I am not too familiar with bbox detection tasks. However, for regression tasks, I think there is a much bigger argument to be made to switch out the language modelling head with a dedicated one. First, the cross-entropy loss typically used with the language modelling head is incorrect for regression tasks and one should instead use something like mean-squared error. Also, for example for reward or value models as common in reinforcement learning from human feedback (RLHF), everyone already does replace the language modelling head with a regression head, even though few people use transformer-heads for this.
